zipline/finance/performance/position_tracker.py

```diff
@@ -104,33 +152,71 @@ def position_values(self):
         self._position_last_sale_prices is changed.
         """
         if self._position_values is None:
-            vals = list(map(mul, self._position_amounts.values(),
-                        self._position_last_sale_prices.values()))
+            # Apply value multipliers to quantities
+            vals = map(mul, self._position_amounts.values(),
+                       self._position_value_multiplier.values())
+            # Apply last sale prices to quantities
+            vals = list(map(mul, vals,
+                            self._position_last_sale_prices.values()))
             self._position_values = vals
         return self._position_values
 
+    @property
+    def position_exposures(self):
+        """
+        Invalidate any time self._position_amounts or
+        self._position_last_sale_prices is changed.
+        """
+        if self._position_exposures is None:
+            # Apply exposure multipliers to quantities
+            vals = map(mul, self._position_amounts.values(),
+                       self._position_exposure_multiplier.values())
+            # Apply last sale prices to quantities
+            vals = list(map(mul, vals,
+                            self._position_last_sale_prices.values()))
+            self._position_exposures = vals
+        return self._position_exposures
+
     def calculate_positions_value(self):
         if len(self.position_values) == 0:
             return np.float64(0)
 
         return sum(self.position_values)
 
+    def calculate_positions_exposure(self):
+        if len(self.position_exposures) == 0:
+            return np.float64(0)
+
+        return sum(self.position_exposures)
+
     def _longs_count(self):
-        return sum(map(lambda x: x > 0, self.position_values))
+        return sum(map(lambda x: x > 0, self.position_exposures))
```

Comment:
Some more timings, because I found this interesting:

```
In [18]: %timeit len(filter(lambda x: x > 50000, y))
100 loops, best of 3: 4.88 ms per loop

In [19]: %timeit sum(map(lambda x: x > 50000, y))
100 loops, best of 3: 8.22 ms per loop

In [20]: %timeit sum(x > 50000 for x in y)
100 loops, best of 3: 4.08 ms per loop

In [21]: %timeit sum(1 for x in y if x > 50000)
100 loops, best of 3: 2.47 ms per loop
```